src/app/ReadClient.cpp

```diff
@@ -586,14 +597,12 @@ CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle && aPayload)
         }
     }
 
-    if (!suppressResponse)
+    if (!suppressResponse && err == CHIP_NO_ERROR)
```

Comment:
Took me just a bit to figure out that our caller is handling sending StatusResponse in the negative case, and transmission of StatusResponse in the positive cases are handled here. Comments here would be welcome.

Reply:
yes, commented, thanks

Reply:
I don't see the comment.